tex2pdf-service/tex2pdf/tex_to_pdf_converters.py

```diff
@@ -773,8 +773,16 @@ def produce_pdf(self, tex_file: str, work_dir: str, in_dir: str, out_dir: str) -
         step = "tex_to_dvi_run"
         run = self._base_to_dvi_run(step, self.stem, args, work_dir, in_dir)
         dvi_size = run["dvi"]["size"]
-        if not dvi_size:
-            outcome.update({"status": "fail", "step": step, "reason": "failed to create pdf", "runs": self.runs})
+        if not dvi_size or run["return_code"] != 0:
+            msg = "failed to create dvi" if not dvi_size else "compiler run returned error code"
+            outcome.update({"status": "fail", "step": step, "reason": msg, "runs": self.runs})
+            dvi_file = os.path.join(in_dir, f"{self.stem}.dvi")
+            if os.path.exists(dvi_file):
+                os.unlink(dvi_file)
+            run["dvi"] = file_props(dvi_file)
+            outcome.update(
+                {"status": "fail", "step": step, "reason": "compiler run returned error code", "runs": self.runs}
+            )
```

Comment:
Is one of the two `outcome.update` calls here redundant?

Reply:
Deleted the second one.